src/main/scala/diplomacy/Nodes.scala

```diff
@@ -558,45 +1375,58 @@ class MixedAdapterNode[DI, UI, EI, BI <: Data, DO, UO, EO, BO <: Data](
     }
   }
   protected[diplomacy] def mapParamsD(n: Int, p: Seq[DI]): Seq[DO] = {
-    require(n == p.size, s"$context has ${p.size} inputs and ${n} outputs; they must match")
+    require(n == p.size, s"$context has ${p.size} inputs and $n outputs; they must match")
     p.map(dFn)
   }
   protected[diplomacy] def mapParamsU(n: Int, p: Seq[UO]): Seq[UI] = {
-    require(n == p.size, s"$context has ${n} inputs and ${p.size} outputs; they must match")
+    require(n == p.size, s"$context has $n inputs and ${p.size} outputs; they must match")
     p.map(uFn)
   }
 }
 
+/** A node which modifies the parameters flowing through it, but without changing the number of edges or the diplomatic protocol implementation. */
 class AdapterNode[D, U, EO, EI, B <: Data](imp: NodeImp[D, U, EO, EI, B])(
   dFn: D => D,
   uFn: U => U)(
   implicit valName: ValName)
     extends MixedAdapterNode[D, U, EI, B, D, U, EO, B](imp, imp)(dFn, uFn)
 
-// IdentityNodes automatically connect their inputs to outputs
+/** A node which does not modify the parameters nor the protocol for edges that pass through it.
+  *
+  * During hardware generation, identityNodes automatically connect their inputs to outputs.
+  */
 class IdentityNode[D, U, EO, EI, B <: Data](imp: NodeImp[D, U, EO, EI, B])()(implicit valName: ValName)
   extends AdapterNode(imp)({ s => s }, { s => s })
 {
   override def description = "identity"
   protected override val identity = true
-  override protected[diplomacy] def instantiate() = {
+  override protected[diplomacy] def instantiate(): Seq[Dangle] = {
     val dangles = super.instantiate()
-    (out zip in) map { case ((o, _), (i, _)) => o <> i }
+    (out zip in) foreach { case ((o, _), (i, _)) => o <> i }
     dangles
-  } 
+  }
 }
 
-// EphemeralNodes disappear from the final node graph
+/** EphemeralNodes are used for temporary connections, but disappear from the final node graph. */
```

Comment:
The connections are not temporary, the node is. An ephemeral node provides a mechanism to directly connect two nodes to each other where neither node knows about the other, but both know about an ephemeral node they can use to facilitate the connection.